### PR TITLE
docs: reorganize README and add testing.md

### PR DESCRIPTION
**Pull request by lance**
This commit starts to reorganize the README document so that it's more
of a high-level overview with links to other documents and information.
Most of the text from the README had been moved to FAQ.md. The tutorial
has been modified to account for some minor changes in CLI usage. Finally
I have added a testing.md document which can be used to add test cases
for Test Day.

Signed-off-by: Lance Ball <lball@redhat.com>

### CODE REVIEW

**Comment by lance** [comment]
To see the changes rendered: https://github.com/boson-project/functions/tree/dev-preview-prep

File: tutorial.md [hunk 1]
```diff
@@ -1,17 +1,24 @@
...
 ## Prerequisites
-In order to follow along with this tutorial, you will need to have a few
-tools installed. This project currently is limited to Linux.
 
-* [Docker](https://docs.docker.com/install/)
-* [oc](https://docs.openshift.com/container-platform/4.2/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands) or [kubectl](https://kubernetes.io/docs/tasks/tools/install-kubectl/) CLI
-* [kn](https://knative.dev/docs/install/install-kn/) CLI
+In order to follow along with this tutorial, you will need to have a few tools
+installed.
+
```
Comment (zroubalik) [inline tutorial.md:10]: these links doesn't work

File: tutorial.md [hunk 2]
```diff
@@ -20,16 +27,20 @@ these [step by step instructions](kind-setup.md) to install on your local
 machine.
 
 ## Boson Tooling
+
 The primary interface for Boson project is the `faas` CLI.
 [Download][faas-download] the most recent version and install it some place
```
Comment (zroubalik) [inline tutorial.md:32]: Just curious why don't you use this syntax?

`[Download](https://github.com/boson-project/faas/releases)`

File: tutorial.md [hunk 3]
```diff
@@ -167,19 +181,19 @@ faas build
...
+### `faas deploy`
+
 To deploy the image to your cluster, use the `deploy` command.
 
 ```bash
 faas deploy
 ```
 
-### faas update
+### `faas update`
```
Comment (zroubalik) [inline tutorial.md]: we don't have `faas update` anymore

Comment (lance) [reply tutorial.md]: Oops! Good point :)

File: tutorial.md (hunk 2 above)
Comment (lance) [reply tutorial.md:32]: The lines were really long with the URLs inline. I was just trying to keep it a little neater in the text.

File: tutorial.md (hunk 1 above)
Comment (lance) [reply tutorial.md:10]: Obviously I don't know how to use markdown! Will fix it. :)

Comment (zroubalik) [reply tutorial.md:10]: No worries, I don't know either :))

File: tutorial.md [hunk 4]
```diff
@@ -1,17 +1,24 @@
...
-* [kn](https://knative.dev/docs/install/install-kn/) CLI
+In order to follow along with this tutorial, you will need to have a few tools
+installed.
+
+* [oc][oc] or [kubectl][kubectl] CLI
+* [kn][kn] CLI
+* [Docker][docker] 
 
+[docker]: https://docs.docker.com/install/
+[oc]: https://docs.openshift.com/container-platform/4.2/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli_cli-developer-commands
```
Comment (zroubalik) [inline tutorial.md]: just a nit, we might want to target OCP 4.6 in the link (I doubt anything was changed there, but you know)

**Comment by lance** [comment]
Landing this and we can iterate on the testing doc next week

**Pull request merged**